src/java/org/apache/cassandra/service/paxos/cleanup/PaxosCleanupSession.java

```diff
@@ -95,49 +93,40 @@ private static ScheduledFuture<?> schedule(PaxosCleanupSession session)
         }
     }
 
-    private final UUID session = UUID.randomUUID();
+    private final SharedContext ctx;
+    public final UUID session = UUID.randomUUID();
```

Comment:
this is non-deterministic, but this problem exists in repair as well... Repair uses random/time uuid outside of the random source/clock from `SharedContext`... since these are keys for a map, I have mostly looked the other way... it makes debugging annoying (as you can not block on the repair id) but its not been an issue so just ignore